handler/handler_test.go

```diff
@@ -126,6 +134,83 @@ func TestContentPublishedHandler_Handle(t *testing.T) {
 	})
 }
 
+func TestHandlerForZebedeeReturingAllFields(t *testing.T) {
+
+	expectedFullSearchDataImportEvent := models.SearchDataImport{
+		DataType:        "testDataType",
+		JobID:           "JobId",
+		SearchIndex:     "ONS",
+		CDID:            "testCDID",
+		DatasetID:       "testDaetasetId",
+		Keywords:        []string{"testkeyword"},
+		MetaDescription: "testMetaDescription",
+		Summary:         "testSummary",
+		ReleaseDate:     "testReleaseDate",
+		Title:           "testTitle",
+	}
+
+	kafkaProducerMock := &kafkatest.IProducerMock{
+		ChannelsFunc: getChannelFunc,
+	}
+
+	marshallerMock := &mock.MarshallerMock{
+		MarshalFunc: func(s interface{}) ([]byte, error) {
+			return schema.SearchDataImportEvent.Marshal(s)
+		},
+	}
+
+	producerMock := &event.SearchDataImportProducer{
+		Producer:   kafkaProducerMock,
+		Marshaller: marshallerMock,
+	}
+
+	Convey("Given an event handler working successfully, and an event containing a URI", t, func() {
+
+		//used by zebedee mock
+		fullContentPublishedTestData := `{"description":{"cdid": "testCDID","datasetId": "testDaetasetId","edition": "testedition","keywords": ["testkeyword"],"metaDescription": "testMetaDescription","releaseDate": "testReleaseDate","summary": "testSummary","title": "testTitle"},"type": "testDataType"}`
+		getFullPublishDataFunc := func(ctx context.Context, uriString string) ([]byte, error) {
+			data := []byte(fullContentPublishedTestData)
+			return data, nil
+		}
+
+		var zebedeeMock = &clientMock.ZebedeeClientMock{GetPublishedDataFunc: getFullPublishDataFunc}
+		eventHandler := &handler.ContentPublishedHandler{zebedeeMock, *producerMock}
+
+		Convey("When given a valid event", func() {
+
+			err := eventHandler.Handle(context.Background(), &config.Config{}, &testEvent)
+
+			var avroBytes []byte
+			var testTimeout = time.Second * 5
+			select {
+			case avroBytes = <-pChannels.Output:
+				log.Event(ctx, "avro byte sent to producer output", log.INFO)
+			case <-time.After(testTimeout):
+				t.Fatalf("failing test due to timing out after %v seconds", testTimeout)
+				t.FailNow()
+			}
+
+			Convey("Then no error is reported", func() {
+				So(err, ShouldBeNil)
+			})
+			Convey("And then get content published from zebedee", func() {
+				So(zebedeeMock.GetPublishedDataCalls(), ShouldNotBeEmpty)
+				So(zebedeeMock.GetPublishedDataCalls(), ShouldHaveLength, 1)
+				So(zebedeeMock.GetPublishedDataCalls()[0].UriString, ShouldEqual, testEvent.URI)
+			})
+			Convey("And then the expected bytes are sent to producer.output", func() {
+				var actual models.SearchDataImport
+				err = schema.SearchDataImportEvent.Unmarshal(avroBytes, &actual)
+				So(err, ShouldBeNil)
+				So(expectedFullSearchDataImportEvent.DataType, ShouldResemble, actual.DataType)
```

Comment:
Again I would suggest replacing `ShouldResemble` with `ShouldEqual`, it is more precise.